### DISCUSSION TITLE
Question: How to hide "Program Files" and "System" from Search?

### DISCUSSION THREAD

**Discussion by murty2** [DustinBrett/daedalOS#440] [Q&A]
I do not want the search results (whether searched from File Explorer's search box or the Search icon on Taskbar) to include "**Program Files**" and "**System**"

How to hide "**Program Files**" and "**System**" from Search?

**Comment by DustinBrett** [comment]
The dynamic search happens in `utils\search.ts` and the static initial json file is built via `scripts\searchIndex.js`. It already filters out searching inside the folders you mentioned, but to also exclude those folders you may need to customize things a bit. Currently I don't exclude any folder from being indexed, only some files and inside some folders.

**Comment by murty2** [comment]
In scripts/searchIndex.js, changed
`createSearchIndex(PUBLIC_PATH);`
to
`createSearchIndex(join(PUBLIC_PATH, '/Users/Public'));`

I still have to figure out how to figure how to exclude those folders in File Explorer

**Reply by DustinBrett** [reply]
If you want to filter from File Explorer you might be able to add it to `SYSTEM_PATHS` or `SYSTEM_FILES` in `utils\constants.ts`.

**Comment by murty2** [comment]
I took the lazy way :) In package.json, I changed
`"build:fs": "node scripts/fs2json.js --exclude .index  --out public/.index/fs.9p.json ./public",`
to
`"build:fs": "node scripts/fs2json.js --exclude .index --exclude System --exclude 'Program Files' --out public/.index/fs.9p.json ./public",`

**Reply by DustinBrett** [reply]
> I took the lazy way :) In package.json, I changed `"build:fs": "node scripts/fs2json.js --exclude .index --out public/.index/fs.9p.json ./public",` to `"build:fs": "node scripts/fs2json.js --exclude .index --exclude System --exclude 'Program Files' --out public/.index/fs.9p.json ./public",`

Nice, if it works then go for it.

**Comment by murty2** [comment]
Adding /favicon.ico to SYSTEM_PATHS for example only hides from Explorer, not Browser directory index

**Reply by DustinBrett** [reply]
To hide from indexing you'd need to make changes in `searchIndex.js` and `search.ts`.
